src/linalg/impl_linalg.rs

```diff
@@ -815,3 +815,129 @@ mod blas_tests {
         assert!(blas_column_major_2d::<f32, _>(&m));
     }
 }
+
+#[allow(dead_code)]
+fn general_outer_to_dyn<Sa, Sb, I, F, T>(
+    a: &ArrayBase<Sa, IxDyn>,
+    b: &ArrayBase<Sb, I>,
+    f: F,
+) -> ArrayD<T>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    I: Dimension,
+    F: Fn(ArrayViewMut<T, IxDyn>, T, &ArrayBase<Sb, I>) -> (),
+{
+    //Iterators on the shapes, compelted by 1s
+    let a_shape_iter = a.shape().iter().chain([1].iter().cycle());
+    let b_shape_iter = b.shape().iter().chain([1].iter().cycle());
+
+    let res_ndim = std::cmp::max(a.ndim(), b.ndim());
+    let res_dim: Vec<Ix> = a_shape_iter
+        .zip(b_shape_iter)
+        .take(res_ndim)
+        .map(|(x, y)| x * y)
+        .collect();
+
+    unsafe {
+        let mut res: ArrayD<T> = ArrayBase::uninitialized(res_dim);
+        let res_chunks = res.exact_chunks_mut(b.shape());
+        //azip!(mut r_c (res_chunks), a in {f(r_c, a, b)});
+        Zip::from(res_chunks).and(a).apply(|r_c, &x| f(r_c, x, b));
+        res
+    }
+}
+
+#[allow(dead_code)]
+fn kron_to_dyn<Sa, I, Sb, T>(a: &ArrayBase<Sa, IxDyn>, b: &ArrayBase<Sb, I>) -> Array<T, IxDyn>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    I: Dimension,
+    T: crate::ScalarOperand,
+    for<'a> &'a ArrayBase<Sb, I>: std::ops::Mul<T, Output = Array<T, I>>,
```

Comment:
This seems inefficient since it's using a multiplication that produces an Array result, can we use a reusable scratch array for this or some other way to make this mostly in-place?

Maybe using Zip to make the assignment directly without using assign

Reply:
Hi !

Yes indeed, this is not the most efficient. Yet, I haven't really found a function allowing me to directly perform a multiplication into a specified array/buffer. A possibility would be to use `MulAssign` but then I would have to add a pass to copy the first matrix and then mulassign each chunk, which I'm not sure would be more efficient.

> Maybe using Zip to make the assignment directly without using assign

Not sure of what you mean here.

Let me know what you think, and thanks for your review.

Reply:
Hey @bluss, I'm allowing myself to bump the discussion, do you have any opinion, let me know if something isn't clear.